### PR TITLE
drivers: wifi: simplelink: Add static IP address support

### LINKED ISSUE #14588: static IP support on cc3220sf_launchxl
**Is your enhancement proposal related to a problem? Please describe.**
Currently, CONFIG_NET_CONFIG_SETTINGS forces hardcoded values for static IP address, and it does not work well for the offload driver for cc3220 which automatically handles DHCP. So the application needs to disable this feature today in their configuration on cc3220. An example of this is in PR #14548.

**Describe the solution you'd like**
We should at a minimum allow the user to enable CONFIG_NET_CONFIG_SETTINGS and set the device's IP address via CONFIG_NET_CONFIG_MY_IPV4_ADDR (and CONFIG_NET_CONFIG_MY_IPV6_ADDR if and when IPv6 is supported)

**Describe alternatives you've considered**
n/a

### PR DESCRIPTION
**Pull request by vanti**
Change the SimpleLink wifi driver to use static IP address instead of
DHCP when it is set by NET_CONFIG_MY_IPV4_ADDR. We also support setting
gateway and netmask via NET_CONFIG_MY_IPV4_GW and
NET_CONFIG_MY_IPV4_NETMASK.

Fixes #14588

Signed-off-by: Vincent Wan <vincent.wan@linaro.org>

### CODE REVIEW

File: drivers/wifi/simplelink/simplelink_support.c [hunk 1]
```diff
@@ -65,6 +65,44 @@ struct sl_connect_state sl_conn;
 /* Network Coprocessor state, including role and connection state: */
 static struct nwp_status nwp;
 
+#if defined(CONFIG_NET_IPV4) && defined(CONFIG_NET_CONFIG_MY_IPV4_ADDR)
+/* Convert IP addr string into an array of integers */
+static int tokenize_ipv4addr(u8_t *ip_array, char const *ip_string)
```
Comment (jukkar) [inline drivers/wifi/simplelink/simplelink_support.c]: You could use `net_ipaddr_parse()` function here that will do basically the same.

**Review by jukkar — CHANGES_REQUESTED** [review]

**Comment by pfalcon** [comment]
It's absolutely great to see this feature implemented (and thus, cc3220 to fit much better with Zephyr network configuration model). But I agree with @jukkar's comment that we should reuse existing routines whenever possible.

File: drivers/wifi/simplelink/simplelink_support.c (hunk 1 above)
Comment (pfalcon) [reply drivers/wifi/simplelink/simplelink_support.c]: > You could use net_ipaddr_parse() function here that will do basically the same.

I guess net_addr_pton() might be even better fit.

**Review by GAnthony — APPROVED** [review]
Could you please mention in the commit how this was tested?
Which samples, and did wifi_shell work as well with auto init?
If this commit is all that was required, then that's great!

**Comment by vanti** [comment]
> Could you please mention in the commit how this was tested?
> Which samples, and did wifi_shell work as well with auto init?
> If this commit is all that was required, then that's great!



> Could you please mention in the commit how this was tested?
> Which samples, and did wifi_shell work as well with auto init?
> If this commit is all that was required, then that's great!

@GAnthony It was tested out-of-tree against the echo sample after modifying the prj.conf file to set the IP address:

`CONFIG_NET_CONFIG_MY_IPV4_ADDR="192.168.1.191"`

and then I had to comment out these lines in the board-specific config file 

```
#CONFIG_NET_CONFIG_SETTINGS=n
#CONFIG_NET_CONFIG_MY_IPV4_ADDR=""
#CONFIG_NET_CONFIG_PEER_IPV4_ADDR=""
```

that would have overridden the NET_CONFIG_SETTINGS.

I also ran the wifi sample to verify it still runs correctly.

I have added this info to the commit message, thanks.

File: drivers/wifi/simplelink/simplelink_support.c (hunk 1 above)
Comment (vanti) [reply drivers/wifi/simplelink/simplelink_support.c]: @jukkar @pfalcon I have updated the code to use net_addr_pton(). It is a good idea.

File: drivers/wifi/simplelink/simplelink_support.c [hunk 2]
```diff
@@ -115,10 +122,57 @@ static s32_t configure_simplelink(void)
 	retval = sl_WlanProfileDel(0xFF);
 	ASSERT_ON_ERROR(retval, WLAN_ERROR);
 
+#if defined(CONFIG_NET_IPV4) && defined(CONFIG_NET_CONFIG_MY_IPV4_ADDR)
+	if (net_addr_pton(AF_INET, CONFIG_NET_CONFIG_MY_IPV4_ADDR, &addr4)
+			< 0) {
+		LOG_ERR("Please verify CONFIG_NET_CONFIG_MY_IPV4_ADDR");
```
Comment (pfalcon) [inline drivers/wifi/simplelink/simplelink_support.c]: Minor, but I'd say, consistent Zephyr error style would be "Invalid CONFIG_NET_CONFIG_MY_IPV4_ADDR".

**Comment by pfalcon** [comment]
> I have added this info to the commit message

Thumbs up to that, it may be very helpful to someone who will debug this stuff later.

**Review by pfalcon — APPROVED** [review]
Thanks for updates.

File: drivers/wifi/simplelink/simplelink_support.c (hunk 2 above)
Comment (vanti) [reply drivers/wifi/simplelink/simplelink_support.c]: Thanks, done.

**Review by jukkar — APPROVED** [review]

**Pull request merged**